File: uri/estructuras/train_swapping/1162.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int n;
    cin >> n;
    
    while (n--) {
        int l;
        cin >> l;
        
        vector<int> vagones(l);
        for (int i = 0; i < l; i++) {
            cin >> vagones[i];
        }
        
        int swaps = 0;
        for (int i = 0; i < l; i++) {
            for (int j = i + 1; j < l; j++) {
                if (vagones[i] > vagones[j]) {
                    swaps++;
                }
            }
        }
        
        cout << "Optimal train swapping takes " << swaps << " swaps.\n";
    }
    
    return 0;
}

File: uri/estructuras/train_swapping/1162.rs
use std::io::{self, BufRead, Write, BufWriter};

fn main() {
    let stdin = io::stdin();
    let stdout = io::stdout();
    let mut out = BufWriter::new(stdout.lock());
    let mut lines = stdin.lock().lines();
    
    let n: i32 = lines.next().unwrap().unwrap().trim().parse().unwrap();
    
    for _ in 0..n {
        let l: usize = lines.next().unwrap().unwrap().trim().parse().unwrap();
        
        let vagones: Vec<i32> = if l > 0 {
            lines.next().unwrap().unwrap()
                .split_whitespace()
                .map(|x| x.parse().unwrap())
                .collect()
        } else {
            vec![]
        };
        
        let mut swaps = 0;
        for i in 0..l {
            for j in (i + 1)..l {
                if vagones[i] > vagones[j] {
                    swaps += 1;
                }
            }
        }
        
        writeln!(out, "Optimal train swapping takes {} swaps.", swaps).unwrap();
    }
}
